Add side dots for carousel navigation in partially open drawer (#605, #733)

Clement-coder opened the pull request:
Added side dots to the photo carousel in EntryCarousel to indicate swipe navigation in a partially open drawer on mobile. This improves UX by providing visible navigation cues when the bottom dots are hidden.

- Added left and right side dots with click handlers for navigation.
- Hid bottom dots on mobile to avoid clutter in partially open drawer.
- Ensured side dots are only shown when there are multiple photos.

Closes #741

wbazant commented:
Thanks! I fixed your PR description - it seems you used a LLM to write the PR, and it claimed to close issues that your solution doesn't actually close.

1. Could you undo changes to files like package.json that this PR contains? Looks like you used npm to install it, and overwrote the setup based on yarn. Try follow the yarn based instructions to set up the project.

2. The new dots don't look good on desktop together with the old dots - can we have the old dots on desktop and new dots on mobile?

3. Can the photos be conceptually in a line, so we can go left and right, but not left when we're on the first photo and not right when we're on the last photo?

4. No need to make the dots themselves bigger, but can you make the click target a bit bigger, so the person using a phone with their thumb can easily aim at the dot and not the photo?

5. Can you use colors consistent with the rest of the app? see src/components/ui/GlobalStyle.js

6. After zooming into a picture, the dots go away and arrows appear instead. Can we offer a consistent way of navigating between the photos?

wbazant commented:
Thanks! No need to use lucid-react, if you want icons we have the boxicons library. What's your thinking behind changing dots to arrows?

wbazant commented:
Thanks! Feel free to take the time to do the changes. I've not seen how the latest version looks yet - I checked your first PR and now looked at the code - but before I do I think this is still to be addressed:

>  Can the photos be conceptually in a line, so we can go left and right, but not left when we're on the first photo and not right when we're on the last photo?

Can we implement this so that the 'go back' indicator is only visible on 2,3,4... photo, taking the user to the photo with one index less, and the 'go forward' indicator similarly? 'go back' is usually left and 'go forward' is right but we also have translations in Arabic and Hebrew where it's the opposite.

Also, see code review comments.

Clement-coder commented:
Fixed carousel navigation limits (no going past first/last image)

Renamed arrows for RTL support (BackArrow, ForwardArrow)

Removed unnecessary dependency (boxicons, etc)

Restored .env config

wbazant commented:
Thanks! Please also restore the yarn.lock file, we don't want those changes in this PR.

I checked out your code and ran it. The changes that only show the dot when there's something to show forward/backward seem to work in an LTR locale like English, thanks!

Can we improve the visual aspect of the solution? It looks like that on desktop, on mobile on half-open drawer, on mobile in fully open drawer, and on mobile in an open carousel:

![Screenshot from 2025-06-03 09-12-22](https://github.com/user-attachments/assets/969a0320-8130-440b-9a01-2ad29913dcbe)
![Screenshot from 2025-06-03 09-13-14](https://github.com/user-attachments/assets/7ab9c864-98e1-4902-a4e5-fe00ab5cbd76)
![Screenshot from 2025-06-03 09-13-40](https://github.com/user-attachments/assets/52924fb2-f623-4c4e-818f-8b90d06e1363)
![Screenshot from 2025-06-03 09-16-28](https://github.com/user-attachments/assets/d922b398-3d1e-48eb-85aa-791176c09f35)

I'm guessing it's just a result of LLM changes as opposed to a design? A designed solution would be nice - maybe it'll help if you draw in Paint or on a piece of paper what you'd want each of these views to look like first.

wbazant commented:
You can get your own for free I think - do the setup instructions mention that variable somewhere?

wbazant commented:
I looked at your code changes - changing PageTemplate.js wasn't right but otherwise I see you tried to improve how the dots look - and I ran your changes locally. Sorry to be closing the PR, but I appreciate the work you did - it helped me realise my suggested solution is probably not right here. Let's solve this issue differently - I've made #827, feel free to try it or take another issue.
